Code/GraphMol/FMCS/testFMCS_Unit.cpp

```diff
@@ -596,55 +598,131 @@ void testSimpleFast() {
-        mols.push_back(mols[i]);
-      }
+        mols.push_back(ROMOL_SPTR(SmilesToMol( getSmilesOnly(smi[i]) )));
+    t0 = nanoClock();
+    p.AtomCompareParameters.MatchChiralTag = true;
+    p.BondCompareParameters.MatchStereo    = true;
+    MCSResult res = findMCS(mols, &p);
+    std::cout << "MCS: "<<res.SmartsString<<" "<< res.NumAtoms<<" atoms, "<<res.NumBonds<<" bonds\n";
+    printTime();
+    //TEST_ASSERT(res.NumAtoms > 0 && res.NumBonds > 0);
+    if(res.NumAtoms > 0 && res.NumBonds > 0) {  //no MCS test
+        BOOST_LOG(rdInfoLog) << "\t*** TEST FAILED ***\n"; //exit(1);
     }
-    for(int iter=0;iter<10;++iter){
-      MCSParameters lp;
-      lp.BondCompareParameters.RingMatchesRingOnly = true;
-      lp.BondCompareParameters.CompleteRingsOnly   = true;
-      MCSResult res = findMCS(mols, &lp);
-      TEST_ASSERT(res.NumAtoms==15 && res.NumBonds==14);
+    else
+        BOOST_LOG(rdInfoLog) << "\tdone" << std::endl;
+}
+
+void compareChirality(const char* target, const char* query, bool useChirality) {
+
+   BOOST_LOG(rdInfoLog) << "-------------------------------------" << std::endl;
+   std::cout << "test target = " << target << " query = " << query << " useChirality = " << useChirality << std::endl;
+   ROMOL_SPTR target_ptr(SmilesToMol(target));
+   ROMOL_SPTR query_ptr(SmartsToMol(query));
+
+   std::vector< std::pair<int, int> > vect;
+   bool sub_res = SubstructMatch(*target_ptr.get(), *query_ptr.get(), vect, true, useChirality);
+
+   std::cout << "res sub = " << sub_res << " size="<<vect.size() << std::endl;
+
+   std::vector<ROMOL_SPTR> mols;
+   mols.push_back(target_ptr);
+   mols.push_back(query_ptr);
+
+   p.AtomCompareParameters.MatchChiralTag = useChirality;
+   p.BondCompareParameters.MatchStereo = useChirality;
+
+   MCSResult mcs_res = findMCS(mols, &p);
+   std::cout << "MCS: " << mcs_res.SmartsString << " " << mcs_res.NumAtoms << " atoms, " << mcs_res.NumBonds << " bonds\n";
+
+   bool mcs_resb = query_ptr->getNumAtoms() == mcs_res.NumAtoms && query_ptr->getNumBonds() == mcs_res.NumBonds;
+   TEST_ASSERT(sub_res == mcs_resb);
+    if(sub_res != mcs_resb){// || vect.size() != mcs_res.NumAtoms) {
+        BOOST_LOG(rdInfoLog) << "mcs_resb="<<mcs_resb<<"\t*** TEST FAILED ***\n"; //exit(1);
     }
-  }
 }
-#include <boost/thread.hpp>  
-void testMultiThread(){
-  BOOST_LOG(rdErrorLog) << "-------------------------------------" << std::endl;
-  BOOST_LOG(rdErrorLog) << "    Test multithreading" << std::endl;
-
-  boost::thread_group tg;
-  std::cerr<<"processing"<<std::endl;
-  unsigned int count=4;
-  for(unsigned int i=0;i<count;++i){
-    std::cerr<<" launch :"<<i<<std::endl;std::cerr.flush();
-    tg.add_thread(new boost::thread(runblock));
-  }
-  tg.join_all();
+
+void testSubMcsChirality(const char* target, const char* query) {
+   compareChirality(target, query, false);
+   compareChirality(target, query, true);
 }
-#else
-void testMultiThread(){
+
+void testChirality() {
```

Comment:
These tests cover the atom stereochemistry well, but don't hit bond stereochemistry at all.